pkg/logs/input/docker/convertor.go

```diff
@@ -0,0 +1,165 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+// +build docker
+
+package docker
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	iParser "github.com/DataDog/datadog-agent/pkg/logs/parser"
+	"regexp"
+)
+
+var (
+	delimiter = []byte{' '}
+	// 2019-05-29T09:26:32.155255473Z
+	timestampMatcher = regexp.MustCompile("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{9}Z")
+	// Escaped CRLF, used for determine empty messages
+	windowsEOFMark     = []byte{'\\', 'r', '\\', 'n'}
+	carriageReturnMark = []byte{'\\', 'r'}
+	lineFeedMark       = []byte{'\\', 'n'}
+)
+
+const (
+	// [8ByteHeaderTimestamp Message]
+	numOfComponents = 2
+
+	// message indices
+	indexOfHeader  = 0
+	indexOfContent = 1
+
+	// 8Byte header indices
+	indexOfTimestamp   = 8
+	indexOfContentSize = 4
+	indexOfStatus      = 0
+)
+
+// Convertor specific for docker logs read from socket.
+type Convertor struct {
+	iParser.Convertor
+}
+
+// Convert converts the input message to Line, if it's partial msg, defaultPrefix will be
+// used as prefix.
+func (c *Convertor) Convert(msg []byte, defaultPrefix iParser.Prefix) *iParser.Line {
+	components := bytes.SplitN(msg, delimiter, numOfComponents)
+	if !c.validateHeader(components[indexOfHeader]) {
+		// take this msg as partial log splitted by upstream.
+		return &iParser.Line{
+			Prefix:  defaultPrefix,
+			Content: msg,
+			Size:    len(msg),
+		}
+	}
+
+	if c.emptyContent(components) {
+		return nil
+	}
+
+	status, timestamp, contentLen := c.parseHeader(components[indexOfHeader])
+	if contentLen <= 0 && status == "" { // take it as tty, which means msg does not contain 8 bytes docker header.
```

Comment:
```suggestion
	if contentLen == 0 && status == "" { // take it as tty, which means msg does not contain 8 bytes docker header.
```

Reply:
Should we add a specific method to check first if there is a docker header ? I think it would be clearer.